### PR TITLE
11 get articles by topic

### PR DESCRIPTION
**Pull request by Ritzahr**

### CODE REVIEW

File: __tests__/nc_news_test.test.js [hunk 1]
```diff
@@ -159,7 +176,15 @@ describe('GET /api/topics', () => {
...
-})
+        test('Responds with a 404 status code, when requesting a topic that doesn\'t exist', () => {
+            return request(app)
+            .get("/api/articles?topic=philosophy")
+            .expect(404)
+            .then((response)=>{
+                expect(response.body.msg).toBe("No articles found")
+            })
+        }); 
+    });
```
Comment (ghost) [inline __tests__/nc_news_test.test.js:187]: I'd expect to see a 200 test for a valid topic that has no articles. I believe the paper topic has no articles linked to it.

**Review by ghost — COMMENTED** [review]
Looks like two tickets in one here. Users is solid. I'd like to see extra testing for a valid topic with no articles, would you expect to see an error there or a 200 empty array?

File: model/nc_news-model.js [hunk 2]
```diff
@@ -17,15 +17,30 @@ exports.selectArticleByID = (article_id) => {
...
-    return db.query(`
-     SELECT articles.*, COUNT(articles.article_id) AS comment_count
-     FROM articles 
-     LEFT JOIN comments 
-     ON articles.article_id = comments.article_id 
-     GROUP BY articles.article_id 
-     ORDER BY created_at DESC
-     ;`).then((result) => {
+exports.selectAllArticles = (topic) => {
+    if (topic === '') {
```
Comment (ghost) [inline model/nc_news-model.js:21]: If topic hasn't been provided, it wouldn't be an empty string it would just be undefined.

File: model/nc_news-model.js [hunk 3]
```diff
@@ -17,15 +17,30 @@ exports.selectArticleByID = (article_id) => {
...
+    ;`
+    let conditionalQuery = `
+    WHERE topic=$1`;
+    if(topic !== undefined) {
+        queryValues.push(topic)
+        mainQuery+=conditionalQuery       
+    }
+    return db.query(mainQuery+secondaryQuery,queryValues).then((result) => {
+       if (result.rows.length===0) {
+        return Promise.reject({status: 404, msg: "No articles found"})
```
Comment (ghost) [inline model/nc_news-model.js:42]: It isn't inherently a problem that there are no articles if the topic is valid.

**Review by ghost — COMMENTED** [review]
Your query is close but needs a minor refactor to account for valid topics with no articles
